mycpp/leaky_mylib.h

```diff
@@ -125,12 +125,28 @@ class Writer : public Obj {
   virtual bool isatty() = 0;
 };
 
+class Buf {
+ public:
+  Buf(char* data, int len) : data_(data), len_(len) {}
+  explicit operator bool() { return len_ == 0; }
+  char* data() { return data_; }
+  bool IsValid() { return len_ != -1; }
+  void Extend(Str* s);
+  void Invalidate();
+ private:
+  friend Str* StrFromBuf(const Buf&);
+  char* data_;
+  int len_;
+};
+
+Str* StrFromBuf(const Buf&);
+Buf AllocBuf(char*, int);
+
 class BufWriter : public Writer {
  public:
   BufWriter()
       : Writer(Tag::FixedSize, kZeroMask, sizeof(BufWriter)),
-        data_(nullptr),
-        len_(0) {
+        buf_(nullptr, 0) {
```

Comment:
Seems like the constructor doesn't need any params, but we can also sort that out in the next PR

Reply:
done